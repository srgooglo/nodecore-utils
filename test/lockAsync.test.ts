import { range } from 'lodash'

import {
  delay,
  lockAsync,
} from '../src'

async function func(val: number) {
  await delay(Math.random() * 10)
  return val
}

describe('lockAsync', () => {
  it('Should retain the array order', async () => {
    const lockedFunc = lockAsync(func)
    const result: number[] = []
    const promises = range(10).map(i => lockedFunc(i).then(r => result.push(r)))
    await Promise.all(promises)
    expect(result).toMatchSnapshot()
  })
})
