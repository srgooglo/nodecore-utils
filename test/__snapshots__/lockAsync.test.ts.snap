// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lockAsync Should retain the array order 1`] = `
Array [
  0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
]
`;
